src/components/Toggle.tsx

```diff
@@ -1,139 +1,121 @@
-  disabled: boolean;
-  checked: boolean;
-  onChange: OnInputChange;
-}> = (props) => (
-  <input
-    sx={{
-      overflow: "hidden",
-      visibility: "hidden",
-      height: 0,
-      opacity: 0,
-      width: 0,
-      position: "absolute",
-      backgroundColor: "transparent",
-      zIndex: -1,
-    }}
-    {...props}
-  />
-);
-const ToggleHandler: FC<{
-  disabled: boolean;
-  checked: boolean;
-}> = ({ disabled, checked, ...rest }) => (
-  <div
-    sx={{
-      width: "1.75rem",
-      height: "0.875rem",
-      borderRadius: "0.875rem",
-      transitionDelay: "0.12s",
-      transitionDuration: "0.2s",
-      transitionProperty: "background, border",
-      transitionTimingFunction: "cubic-bezier(0, 0, 0.2, 1)",
-      position: "relative",
-      border: "1px solid transparent",
-      backgroundColor: "lightGray",
-      padding: 0,
-      ...(checked
-        ? {
-            backgroundColor: "primary",
-          }
-        : {}),
-      ...(disabled
-        ? {
-            borderColor: "darkGray",
-          }
-        : {}),
-
-      ...(disabled && checked
-        ? {
-            backgroundColor: "lightGray",
-          }
-        : {}),
-    }}
-    {...rest}
-  >
-    <span
+export const Body: FC<
+  CheckDisable & { onChange: (e: ChangeEvent<HTMLInputElement>) => void }
+> = ({ checked, disabled, children, onChange }) => (
+  <Fragment>
+    <input
+      type="checkbox"
+      disabled={disabled}
+      checked={checked}
+      onChange={onChange}
       sx={{
-        width: "calc(0.875rem - 3px)",
-        height: "calc(0.875rem - 3px)",
-        position: "absolute",
-        top: "50%",
-        transform: "translateY(-50%)",
-        left: "1px",
-        boxShadow:
-          "rgba(0, 0, 0, 0.2) 0 1px 2px 0, rgba(0, 0, 0, 0.2) 0 1px 3px 0",
-        transition: "left 280ms cubic-bezier(0, 0, 0.2, 1)",
-        borderRadius: "50%",
-        backgroundColor: "background",
-        ...(checked
-          ? {
-              left: "calc(100% - (0.875rem - 2px))",
-              boxShadow: "none",
-            }
-          : {}),
-        ...(disabled
-          ? {
-              backgroundColor: "darkGray",
-              boxShadow: "none",
-            }
-          : {}),
+        visibility: 'hidden',
+        position: 'absolute',
+        height: 0,
+        width: 0,
+        opacity: 0,
+        overflow: 'hidden',
+        backgroundColor: 'transparent',
+        zIndex: -1,
       }}
     />
-  </div>
-);
+    <div
+      sx={{
+        width: '1.75rem',
+        height: '0.875rem',
+        borderRadius: '0.875rem',
+        transitionDelay: '0.12s',
+        transitionDuration: '0.2s',
+        transitionProperty: 'background, border',
+        transitionTimingFunction: 'cubic-bezier(0, 0, 0.2, 1)',
+        position: 'relative',
+        border: '1px solid transparent',
+        backgroundColor: 'lightGray',
+        padding: 0,
+        ...ifStyle(disabled, { borderColor: 'darkGray' }),
```

Comment:
this one I will call it `ifProp` anyway I still don't like the spread

Reply:
I don't like spreading too.. let's think about it...